include/rmm/cuda_device.hpp

```diff
@@ -102,6 +103,52 @@ inline int get_num_cuda_devices()
   return num_dev;
 }
 
+/**
+ * @brief Returns the available and total device memory in bytes for the current device
+ *
+ * @return The available and total device memory in bytes for the current device as a std::pair.
+ */
+inline std::pair<std::size_t, std::size_t> available_device_memory()
+{
+  std::size_t free{};
+  std::size_t total{};
+  RMM_CUDA_TRY(cudaMemGetInfo(&free, &total));
+  return {free, total};
+}
+
+namespace detail {
+
+/**
+ * @brief Returns the available and total device memory in bytes for the current device
+ *
+ * @deprecated Use rmm::available_device_memory instead.
+ *
+ * @return The available and total device memory in bytes for the current device as a std::pair.
+ */
+[[deprecated("Use `rmm::available_device_memory` instead.")]]  //
+const auto available_device_memory = rmm::available_device_memory;
+
+}  // namespace detail
+
+/**
+ * @brief Returns the approximate specified percent of free device memory on the current CUDA
+ * device, aligned to the nearest CUDA allocation size.
+ *
+ * @param percent The percent of free memory to return. Defaults to 50%.
```

Comment:
**issue** This docstring is at odds with the implementation, which returns the requested percentage of _total_ device memory (or the remaining free memory if that number is too large). I think we want the implementation as is, but then docstring wants changed.

Perhaps

> Return the approximate specified percent of total device memory on the current CUDA device or the total free device memory (whichever is smaller), ...

?

Reply:
Good catch, thanks.